data/persistence/src/commonMain/kotlin/social/androiddev/common/persistence/localstorage/DodoAuthStorageImpl.kt

```diff
@@ -31,29 +33,31 @@ internal class DodoAuthStorageImpl(
      * The user can set up multiple accounts on their device. So we
      * key the AccessToken by the unique server/domain
      */
-    private var diskCache: Map<String, AccessToken>
+    private var diskCache: LinkedHashMap<String, AccessToken>
         get() {
             return settings
                 .getStringOrNull(KEY_ACCESS_TOKENS_CACHE)
                 ?.let { str ->
                     json.decodeFromString(ListSerializer(AccessToken.serializer()), str)
-                        .associateBy { it.server }
+                        .associateByTo(linkedMapOf()) { it.server }
                 }
-                ?: mutableMapOf()
+                ?: linkedMapOf()
         }
         set(value) {
             val list = value.map { it.value }
             settings[KEY_ACCESS_TOKENS_CACHE] =
                 json.encodeToString(ListSerializer(AccessToken.serializer()), list)
         }
+    private val memCache: LinkedHashMap<String, AccessToken> by lazy { diskCache }
 
-    private val memCache: MutableMap<String, AccessToken> by lazy { diskCache.toMutableMap() }
-
-    override fun getAccessToken(server: String): String? = memCache[server]?.token
+    override suspend fun getAccessToken(server: String): String? =
```

Comment:
It appears that `suspend` is used only because of `Mutex`, which makes your calls async without a reason. You may try [atomicfu Locks](https://github.com/Kotlin/kotlinx-atomicfu#locks) instead, or write your own with expect/actual (it's pretty easy).

Reply:
Yeah, that does not seems right to have `suspend`able just because of that. Going to add atomicfu, I think it could come useful later on as well